### PR TITLE
Update readme with latest documentation and diagrams

### PR DESCRIPTION
**Pull request by mcieplak**
@prabushyam

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,76 +1,205 @@
...
-- [Usage](#usage)
+- [References](#references)
 - [Contribute](#contribute)
 - [License](#license)
 
 ## Background
-We needed a controller that can dynamically fetch Athenz role / policy mappings and convert them to their corresponding 
-Istio custom resources, so we built this controller to allow users to define RBAC through Athenz and have it integrate 
-with the Istio world.
+We needed a controller that can dynamically fetch Athenz role / policy mappings
```
Comment (prabushyam) [inline README.md]: > In order to adopt Athenz as an unified RBAC provider for managing access to Kubernetes Services, we needed a controller ...

File: README.md [hunk 2]
```diff
@@ -1,76 +1,205 @@
...
+and convert them to their corresponding Istio custom resources, so we built this
+controller to allow users to define RBAC through Athenz and have it integrate with
+the Istio world.
 
-The role name in Athenz specifies the target service in the ServiceRole object and its corresponding HTTP methods are 
-listed as policies. The services that can access this target service are members of the role and will be listed in the 
-ServiceRoleBinding.
+### Istio RBAC
```
Comment (prabushyam) [inline README.md]: can we include a couple of lines about Athenz similar to the athenz-identity repo: https://github.com/yahoo/k8s-athenz-identity#athenz
> Athenz is an IAM provider which allows users to define role based access control (RBAC) for users and services. It also acts as a certificate authority (CA) by provisioning instances with unique identities through X.509 certificates.

File: README.md [hunk 3]
```diff
@@ -1,76 +1,205 @@
...
+#### Onboarding Controller
+The onboarding controller is responsible for syncing the onboarding state of each
+application. The following steps are taken to sync.
+1. Any update in a service object will trigger an addition onto the work queue if
+an authorization annotation is set to true.
+2. The sync will update the cluster rbac config list with the onboarded service
+to enable authorization from an Istio perspective.
+3. This custom resource is offloaded onto the Istio CR work queue.
+
+To onboard a service to be dynamically synced through this controller, a service
```
Comment (prabushyam) [inline README.md]: Could we mention that 
> The onboarding controller assumes that the ClusterRbacConfig is configured with `Mode: v1alpha1.RbacConfig_ON_WITH_INCLUSION` and the services that require authorization enabled are added individually as FQDNs. The controller does not support any other mode at this point.

**Comment by mcieplak** [comment]
@prabushyam Updated based on your comments, please review. Thanks!

**Pull request merged**